sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -1043,3 +1044,45 @@ def some_pipeline():
       parameter_arguments_json = template['metadata']['annotations']['pipelines.kubeflow.org/arguments.parameters']
       parameter_arguments = json.loads(parameter_arguments_json)
       self.assertEqual(set(parameter_arguments.keys()), {'Input 1'})
+
+  def test_keyword_only_argument_for_pipeline_func(self):
+    def some_pipeline(casual_argument: str, *, keyword_only_argument: str):
+      pass
+    kfp.compiler.Compiler()._create_workflow(some_pipeline)
+
+  def test_keyword_only_argument_for_pipeline_func_identity(self):
+    test_data_dir = os.path.join(os.path.dirname(__file__), 'testdata')
+    sys.path.append(test_data_dir)
+    import basic
+
+    pipeline_func_arg = basic.save_most_frequent_word
```

Comment:
Maybe just define a simple pipeline functions which uses keyword-only parameters and then try to compile it?

Reply:
Sure! I thought using a fixture-like pipeline would be more realistic, but it shouldn't matter in this case. Added.